### PR TITLE
[Validator] Make it explicit when English translation differs from its resource name

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Inspired by #53505
Follows #53420

### CODE REVIEW

**Comment by k0d3r1s** [comment]
i kinda don't get this one. isn't target supposed to be somewhat exact translation of source?  
why only english can differ in this regard? doesn't this make it somewhat harder to maintain? 
also, regarding #53505, for example the one with ip address translation - source is "This value is not valid ...", rename part is "This is not valid ...". What does latvian part be? With or without "value"? 
Technically with as target is directly translating source? Or latvian part may differ from source, as "value" part is as unnecessary part of that sentence as it is in english?

**Comment by nicolas-grekas** [comment]
@k0d3r1s you might want to check #53420, that should answer your concern.

**Comment by k0d3r1s** [comment]
@nicolas-grekas ok, read through it. now i just feel that this needs to be documented somewhere (probably in docs). as now that message in code differs from translation key, it is harder to find why.  
but again then, why is it not consistent? unit 37 (ip address one) has this new rename but unit 38, for example, does not

**Comment by nicolas-grekas** [comment]
Unit 38 already starts with "This value is not" so I just aligned unit 37. This can be documented for sure, that's what I'm trying to do here, just with a technical PoV. Docs improvement would be nice indeed! Help wanted :)
Why isn't this consistent at the source level? History :man_shrugging: And it's hard to remember when reviewing new messages (but we should still be more careful.)

File: src/Symfony/Component/Validator/Resources/translations/validators.af.xlf [hunk 1]
```diff
@@ -134,8 +134,8 @@
                 <source>This file is not a valid image.</source>
                 <target>Hierdie lêer is nie 'n geldige beeld nie.</target>
             </trans-unit>
-            <trans-unit id="37">
-                <source>This is not a valid IP address.</source>
+            <trans-unit id="37" rename="This is not a valid IP address.">
```
Comment (stof) [inline src/Symfony/Component/Validator/Resources/translations/validators.af.xlf]:
```suggestion
            <trans-unit id="37" resname="This is not a valid IP address.">
```

Comment (stof) [reply src/Symfony/Component/Validator/Resources/translations/validators.af.xlf]: the same mistake is done in many other places

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by stof** [comment]
Shouldn't this be implemented by `.github/sync-translations.php` to ensure this is kept uptodate as well ?

The current typos (which are not done for all keys) tell me that this was done manually.

**Comment by nicolas-grekas** [comment]
> Shouldn't this be implemented by .github/sync-translations.php to ensure this is kept uptodate as well ?

yes! script updated.

Status: needs review

**Pull request merged**